lib/destination/dml/columns.go

```diff
@@ -21,7 +21,7 @@ func quoteColumns(cols []columns.Column, dialect sql.Dialect) []string {
 
 func removeDeleteColumnMarker(cols []columns.Column) ([]columns.Column, bool) {
 	origLength := len(cols)
-	cols = slices.DeleteFunc(cols, func(col columns.Column) bool { return col.Name() == constants.DeleteColumnMarker })
+	cols = slices.DeleteFunc(slices.Clone(cols), func(col columns.Column) bool { return col.Name() == constants.DeleteColumnMarker })
```

Comment:
`DeleteFunc` mutates it's inputs. 😬

Reply:
eek

Reply:
Do you want to leave a comment to explain the rationale for `slices.Clone(...)`?